x/incentive/types/params.go

```diff
@@ -0,0 +1,134 @@
+var (
+	KeyActive                = []byte("Active")
+	KeyRewards               = []byte("Rewards")
+	DefaultActive            = false
+	DefaultRewards           = Rewards{}
+	DefaultPreviousBlockTime = tmtime.Canonical(time.Unix(0, 0))
+	GovDenom                 = cdptypes.DefaultGovDenom
+	PrincipalDenom           = "usdx"
+	IncentiveMacc            = kavadistTypes.ModuleName
+)
+
+// Params governance parameters for the incentive module
+type Params struct {
+	Active  bool    `json:"active" yaml:"active"` // top level governance switch to disable all rewards
+	Rewards Rewards `json:"rewards" yaml:"rewards"`
+}
+
+// NewParams returns a new params object
+func NewParams(active bool, rewards Rewards) Params {
+	return Params{
+		Active:  active,
+		Rewards: rewards,
+	}
+}
+
+// DefaultParams returns default params for kavadist module
+func DefaultParams() Params {
+	return NewParams(DefaultActive, DefaultRewards)
+}
+
+// String implements fmt.Stringer
+func (p Params) String() string {
+	return fmt.Sprintf(`Params:
+	Active: %t
+	Rewards: %s`, p.Active, p.Rewards)
+}
+
+// ParamKeyTable Key declaration for parameters
+func ParamKeyTable() params.KeyTable {
+	return params.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// ParamSetPairs implements the ParamSet interface and returns all the key/value pairs
+func (p *Params) ParamSetPairs() params.ParamSetPairs {
+	return params.ParamSetPairs{
+		{Key: KeyActive, Value: &p.Active},
+		{Key: KeyRewards, Value: &p.Rewards},
+	}
+}
+
+// Validate checks that the parameters have valid values.
+func (p Params) Validate() error {
+	rewardDenoms := make(map[string]bool)
+
+	for _, reward := range p.Rewards {
+		if rewardDenoms[reward.Denom] {
+			return fmt.Errorf("cannot have duplicate reward denoms: %s", reward.Denom)
+		}
+		rewardDenoms[reward.Denom] = true
+		if !reward.Reward.IsPositive() {
+			return fmt.Errorf("reward must be positive, is %s for %s", reward.Reward, reward.Denom)
+		}
+		if int(reward.Duration.Seconds()) <= 0 {
+			return fmt.Errorf("reward duration must be positive, is %s for %s", reward.Duration.String(), reward.Denom)
+		}
+		if int(reward.TimeLock.Seconds()) < 0 {
+			return fmt.Errorf("reward timelock must be non-negative, is %s for %s", reward.TimeLock.String(), reward.Denom)
+		}
+		if int(reward.ClaimDuration.Seconds()) <= 0 {
+			return fmt.Errorf("reward timelock must be positive, is %s for %s", reward.ClaimDuration.String(), reward.Denom)
+		}
+	}
+	return nil
+}
+
+// Reward stores the specified state for a single reward period.
+type Reward struct {
+	Active        bool          `json:"active" yaml:"actlive"`                // governance switch to disable a period
+	Denom         string        `json:"denom" yaml:"denom"`                   // the collateral denom rewards apply to, must be found in the cdp collaterals
+	Reward        sdk.Coin      `json:"reward" yaml:"reward"`                 // the total amount of coins distributed per period
```

Comment:
```suggestion
	TotalAvailableRewards        sdk.Coin      `json:"total_available_rewards" yaml:"total_available_rewards"`                 // the total amount of coins distributed per period
```

Bit confusing to have Reward as a field on `Reward`. `TotalAvailableRewards` is just an example, can replace with whatever.

Reply:
This supersedes https://github.com/Kava-Labs/kava/pull/399#discussion_r398112486